### ISSUE TITLE
Can not build td_example.cpp

### ISSUE THREAD

**Issue by min4er** [tdlib/td#113]
Hi, I can not build td_example.cpp, which is offered as example. Could you give some manual how to build it? Do I need include some *dll or *lib resources etc..

**Comment by isopen** [comment]
You need to compile and install tdlib.

**Comment by hekkup** [comment]
## How to Build C++ Example
  
- When configuring TDLib, set the installation folder with -DCMAKE_INSTALL_PREFIX=\<LIBDIR\> parameter. By default the installation folder is something like C:\Program Files(x86)\TdLib\lib. Installing into the default folder requires administrator privileges so use for example -DCMAKE_INSTALL_PREFIX=../lib-cpp/ if you want to avoid admin hassle.

```
cd <TDLIB_ROOT>
mkdir build
cd build
cmake -DCMAKE_INSTALL_PREFIX=<LIBDIR> -DCMAKE_BUILD_TYPE=Release ..
cmake --build . --target install
```

- Build the example by giving the library folder path as CMake parameter -DTd_DIR=\<LIBDIR\>:

```
cd <TDLIB_ROOT>
cd example/cpp
mkdir build
cd build
cmake -DCMAKE_BUILD_TYPE=Release -DTd_DIR=<LIBDIR>/lib/cmake/Td/ ..
cmake --build .
```

- Copy ZLib DLL from VCPkg folder into the folder where the example EXE files reside. For example:

```
cd <VCPKG_DIR>/installed/x86-windows/debug/bin/
cp zlibd1.dll <TDLIB_ROOT>/example/cpp/build/Debug
```

- Now you can run ./td_example.exe and ./tdjson_example.exe.

- Could you please add these instructions as README in CPP folder? Or should I do it and make a pull request?

**Comment by levlam** [comment]
Thanks, @hekkup. We already added README.md to cpp examples with similar instructions. It will be published as part of 1.2.0 release soon.

**Comment by ras0219-msft** [comment]
If you use the vcpkg toolchain file, it will automatically copy every needed DLL out:
```cmake
cmake -DCMAKE_BUILD_TYPE=Release -DTd_DIR=<LIBDIR>/lib/cmake/Td/ .. -DCMAKE_TOOLCHAIN_FILE=/path/to/vcpkg/scripts/buildsystems/vcpkg.cmake
```

https://github.com/Microsoft/vcpkg/blob/master/docs/users/integration.md#cmake-toolchain-file-recommended-for-open-source-cmake-projects

**Comment by hekkup** [comment]
So you have some private repo/branches... Any chance having them in the open? At the very least it would be nice if you published some kind of info about what's going on, in order to avoid duplicate work in open/private branches (seems not to be a big problem at the moment, though).

**Comment by levlam** [comment]
Yes. It can be treated like dev branch. We working there mostly on new features, which uses yet unstable server API and may not work at all, so that branch is generally unusable. Changes are made public when new server API layer became stable. Some of the work not related to the new API, like bug fixing and documentation improvements, may be done in public, but this will require a lot of changes in the development process.

**Comment by levlam** [comment]
C++ [README.md](example/cpp/README.md) with build instructions has been released as part of TDLib 1.2.0.
